src/prefect/cli/deploy.py

```diff
@@ -864,17 +863,20 @@ async def _run_single_deploy(
 
 
 async def _version_info_from_options(
-    options: dict[str, Any], deploy_config: dict[str, Any]
+    client: "PrefectClient",
+    options: dict[str, Any],
+    deploy_config: dict[str, Any],
 ) -> VersionInfo | None:
-    if version_type := (
-        options.get("version_type") or deploy_config.get("version_type")
-    ):
-        app.console.print(f"Inferring version info for {version_type}...")
-        return await get_inferred_version_info(version_type)
+    if client.server_type != ServerType.CLOUD:
+        return None
```

Comment:
Rather than trying to skirt around the issue, I'm just going to restrict this to Prefect Cloud clients, since some versions of Prefect servers don't support the `version_info` attribute.